Src/libCZI_UnitTests/test_azureblobstream.cpp

```diff
@@ -0,0 +1,187 @@
+
+using namespace libCZI;
+using namespace std;
+
+struct AzureUriAndExpectedResultFixture : public testing::TestWithParam<tuple<wstring, map<wstring, wstring>>> { };
+
+TEST_P(AzureUriAndExpectedResultFixture, TokenizeAzureUriScheme_ValidCases)
+{
+    const auto parameters = GetParam();
+    const auto tokens = Utilities::TokenizeAzureUriString(get<0>(parameters));
+    EXPECT_EQ(tokens, get<1>(parameters));
+}
+
+INSTANTIATE_TEST_SUITE_P(
+    AzureBlobStream,
+    AzureUriAndExpectedResultFixture,
+    testing::Values(
+    // third tile on top, second tile in the middle, first tile at the bottom
+    make_tuple(L"a=b;c=d;d=e", map<wstring, wstring> { { L"a", L"b" }, { L"c", L"d" }, { L"d", L"e" } }),
+    make_tuple(L"a=b x;c= d ;d=e  ;o=   w", map<wstring, wstring> { { L"a", L"b x" }, { L"c", L" d " }, { L"d", L"e  " }, { L"o", L"   w" } }),
+    make_tuple(L" a =b;c =d; d =e", map<wstring, wstring> { { L" a ", L"b" }, { L"c ", L"d" }, { L" d ", L"e" } }),
+    make_tuple(LR"(a=\;\;;c=\\d;d=e)", map<wstring, wstring> { { L"a", L";;" }, { L"c", LR"(\\d)" }, { L"d", L"e" } }),
+    make_tuple(LR"(c=\\d)", map<wstring, wstring> { { L"c", LR"(\\d)" } }),
+    make_tuple(LR"(\;a=abc)", map<wstring, wstring> { { LR"(;a)", L"abc" } }),
+    make_tuple(LR"(\;a\==abc)", map<wstring, wstring> { { LR"(;a=)", L"abc" } }),
+    make_tuple(LR"(c=\\d\=\;)", map<wstring, wstring> { { L"c", LR"(\\d=;)" } })
+));
+
+struct IllFormedAzureUriAndExpectedErrorFixture : public testing::TestWithParam<wstring> { };
+
+TEST_P(IllFormedAzureUriAndExpectedErrorFixture, TokenizeAzureUriScheme_InvalidCases)
+{
+    const auto parameter = GetParam();
+    EXPECT_THROW(Utilities::TokenizeAzureUriString(parameter), std::invalid_argument);
+}
+
+INSTANTIATE_TEST_SUITE_P(
+    AzureBlobStream,
+    IllFormedAzureUriAndExpectedErrorFixture,
+    testing::Values(
+    L"xxx",
+    L"=xxx",
+    LR"(a\=xxx)",
+    L";",
+    L"=",
+    L"=;",
+    LR"(\=\;)",
+    L"a=b;c=d;k="
+));
+
+static bool IsAzureBlobInputStreamAvailable()
+{
+    StreamsFactory::CreateStreamInfo create_info;
+    create_info.class_name = "azure_blob_inputstream";
+
+    for (int i = 0; i < StreamsFactory::GetStreamClassesCount(); ++i)
+    {
+        StreamsFactory::StreamClassInfo info;
+        if (StreamsFactory::GetStreamInfoForClass(i, info) && info.class_name == create_info.class_name)
+        {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+static string EscapeForUri(const char* str)
+{
+    string result;
+    for (const char* p = str; *p; ++p)
+    {
+        if (*p == ';')
+        {
+            result += "\\;";
+        }
+        else if (*p == '=')
+        {
+            result += "\\=";
+        }
+        else
+        {
+            result += *p;
+        }
+    }
+
+    return result;
+}
+
+static const char* GetAzureBlobStoreConnectionString()
+{
+    // We use the environment variable 'AZURE_BLOB_STORE_CONNECTION_STRING' to communicate a connection string.
```

Comment:
Is this still relevant since function is returning literal value?

Reply:
Could attempt to read env value and use literal value as fall-back

Reply:
Thanks, good catch - this was plain wrong. The idea here is: read the environment-variable, if present, do the test, if not - skip the test. This is left for the case "Azurite is active, but I don't want to set an environment variable - i.e. for local development".
Not a perfect solution I know, but fits the bill for me.